mail_debrand/models/mail_thread.py

```diff
@@ -10,5 +10,14 @@ class MailThread(models.AbstractModel):
 
     def _replace_local_links(self, html, base_url=None):
         html = super()._replace_local_links(html, base_url=base_url)
-        html_debranded = self.env["mail.template"]._debrand_body(html)
+        lang = False
+        if {'default_template_id', 'default_model', 'default_res_id'} \
+                <= self.env.context.keys():
+            template = self.env["mail.template"].browse(
+                self.env.context['default_template_id'])
+            template_lang = template.lang or template._context.get('lang')
+            lang = template._render_template(template_lang,
```

Comment:
Please only render the language if it has a value as suggested in https://github.com/OCA/social/pull/736#discussion_r778752727.
Moreover, why is this rendering the language coming from the context?

Reply:
From [https://github.com/OCA/social/pull/736#discussion_r733688343](https://github.com/OCA/social/pull/736#discussion_r733688343):

> 
In order to debrand correctly the mail, you must translate the terms `using` and `Powered by` with the language set in the mail.template record **if exists, otherwise the language in context**.

See also [https://github.com/odoo/odoo/blob/2be47637dd5cc48ca561ff85cf2d359dde0829ab/addons/mail/models/mail_template.py#L443](https://github.com/odoo/odoo/blob/2be47637dd5cc48ca561ff85cf2d359dde0829ab/addons/mail/models/mail_template.py#L443)